lib/ansible/modules/network/nxos/nxos_nxapi.py

```diff
@@ -279,6 +286,10 @@ def main():
                            supports_check_mode=True)
 
     warnings = list()
+    warning_msg = "Module nxos_nxapi defaults to configure 'http port 80'."
+    warning_msg += "\nDefault behavior is changing in release 2.11 to configure 'https port 443'"
+    warning_msg += " when params 'http, http_port, https, https_port' are not set in the playbook."
+    warnings.append(warning_msg)
```

Comment:
@mikewiebe The way to do this is to use `module.deprecate`
```
if condition:
    module.deprecate(msg='deprecation msg",
                                  version="2.11")
```
This will generate standard deprecation warning like Ansible does.
I think you can actually eliminate the condition part here as default is set in argspec.